code/modules/mining/smelting.dm

```diff
@@ -1,142 +1,125 @@
 /datum/smelting_recipe
-	var/name=""
-	var/list/ingredients=list() // phazon=1, iron=1
-	var/yieldtype=null
+	var/name= ""
+	var/list/ingredients[0]
+	var/yieldtype = null
 
 // Note: Returns -1 if not enough ore!
 /datum/smelting_recipe/proc/checkIngredients(var/obj/machinery/mineral/processing_unit/P)
-	var/sufficient_ore=1
-	var/matching_ingredient_count=0
 	for(var/ore_id in P.ore.storage)
-		var/required=(ore_id in ingredients)
-		var/selected=(ore_id in P.selected)
+		var/min_ore_required = ingredients[ore_id]
 
-		// Selected but not in ingredients
-		if(selected && !required)
+		if(P.ore.getAmount(ore_id) < min_ore_required)
 			return 0
 
-		// Unselected but in ingredients
-		if(!selected && required)
-			return 0
-
-		var/min_ore_required=ingredients[ore_id]
-
-		// Selected, in ingredients, but not enough in stock.
-		if(selected && required)
-			if(P.ore.storage[ore_id] < min_ore_required)
-				sufficient_ore=0
-				continue
-
-			matching_ingredient_count++
-
-	if(!sufficient_ore)
-		return -1 // -1 means not enough ore. NOT A TYPO.
-
-	return matching_ingredient_count == ingredients.len
+	. = 1
 
 // RECIPES BEEP BOOP
 /datum/smelting_recipe/glass
-	name="Glass"
+	name = "Glass"
 	ingredients=list(
-		MAT_GLASS=1
+		MAT_GLASS = 1
 	)
-	yieldtype=/obj/item/stack/sheet/glass/glass
+	yieldtype = /obj/item/stack/sheet/glass/glass
 
+/*
```

Comment:
I could be wrong, but I think it works the other way, too
You get a full metal sheet from smelting a rod
...Maybe